### ISSUE TITLE
Asus Zephyrus Duo / (maybe also zenbook duo) Secondary Screen Toggle

### ISSUE THREAD

**Issue by squee666** [seerge/g-helper#1215]
I have a Zephyrus Duo 2023 and Ghelper is really useful for it.

Only downside is lack of 2nd screen control when asus services are disabled.

It has a dedicated Key for controlling the display via one of the services ghelper disables in app

Here is a image of what the osd looks like when toggled

and heres an example image of the key itself froma  zeph duo 15 inch

https://i.imgur.com/RNb1IGO.png

**Comment by seerge** [comment]
@squee666 you actually can control second screen brightness (including turning it off completely) in g-helper. You can bind  macro-keys to "Screenpad brightness" in Extra settings.

Check ``%appdata%\ghelper\log.txt`` for what exact keycode does that special key produce (as soon as you press it), i can add an extra binding for it too.

**Comment by squee666** [comment]
ahh thats smart
the Keycode is 106 btw

but yeah I would rather just have it Function exactly like it does here

https://i.imgur.com/Cx9jRKQ.png

Basically it just need to Toggle between 3 functions, 
Display On
Disaply backlight off
Display off entirely ( as its physically disconnected aswell from display settings)

**Comment by seerge** [comment]
@squee666 what does "backlight off" mean? is it just minimal brightness ? (try macros from Extra, when you set brightness to 0 it would also turn display OFF in the app)

**Comment by squee666** [comment]
> @squee666 what does "backlight off" mean? is it just minimal brightness ? (try macros from Extra, when you set brightness to 0 it would also turn display OFF in the app)

Yeah your right that turns off display entirely at 0% but the dedicated button functions like this

Display ON 100% Brightness
Display 0% Brightness / Touch disabled but its shows as still connected and windows can still be dragged to it etc
Display OFF


i dont know how asus is doing it but hide display option in OSD turns off the backlight and touch input 
Basically its acting like a second monitor is still plugged into the input and detected but is in a forma of standby

https://i.imgur.com/5BqdCgh.png

**Comment by seerge** [comment]
@squee666 check this build

1. Screenpad UP / Down bindings will also allow you to set brightness to 0% (but keep it off), next step will turn it  off
2. ``106`` code will toggle between OFF / 0% / 100% states (only for case when you have asus services OFF, otherwise it will not interfere)

[GHelper.zip](https://github.com/seerge/g-helper/files/12474340/GHelper.zip)

**Comment by squee666** [comment]
> @squee666 check this build
> 
> 1. Screenpad UP / Down bindings will also allow you to set brightness to 0% (but keep it off), next step will turn it  off
> 2. `106` code will toggle between OFF / 0% / 100% states (only for case when you have asus services OFF, otherwise it will not interfere)
> 
> [GHelper.zip](https://github.com/seerge/g-helper/files/12474340/GHelper.zip)

okay new major issue, the display is disabled  with no touch and image.

it doesnt renable is effectively locked on hidden.
i cant turn it back on now in ghelper or armoury crate but shows as connected

**Comment by seerge** [comment]
@squee666 do you mean that display doesn't enable after getting disabled ? check this build

[GHelper.zip](https://github.com/seerge/g-helper/files/12474792/GHelper.zip)

**Comment by squee666** [comment]
@seerge 

if i use the versions you provided toggling brightness instantly sets it to 100 and disables the display
toggling any of the 10% levels from 0-100 doesnt reanable the backlight or the touch.
this will persist through turning the display off entirely through the hotkey same for using the hotkey for brightness controls

i have to spam the hotkey button to turn the 2nd display on / off until it will eventually kick into 100% brightness.
once its on if i toggle any brightness settings via keycombo Screenpad Down/Up
it will instantly disable backlight and touch.

however if i renable the screen kill any of the previous 2 ghelper builds and use official build 117 brightness toggle functions as it should from 100-0- screen off entirely

but then it doesnt have the function that turns of 2nd screen backlight and touch

heres an example video by what i mean (dont mind the fingerprints hair just had a dog abuse me)
as you can see what i mean by spamming, this doesnt fix it in video but eventually works after i stop recording

**Comment by seerge** [comment]
@squee666 i don't see video, but check this build

[GHelper.zip](https://github.com/seerge/g-helper/files/12475045/GHelper.zip)

**Comment by squee666** [comment]
https://youtu.be/-wFQ9SL-syA

this

i will test build now

**Comment by squee666** [comment]
right test it and it works bur i think asus set the logic like this because otherwise brightness 0 screenpad on doesnt work

Screenpad ON, Brightness 100%
Screenpad ON, Brightness 0%
Screenpad OFF, Brightness OFF

**Comment by seerge** [comment]
@squee666 ok, then brightness 0 is not the right value. Lets try 1 instead of 0

Check this build
[GHelper.zip](https://github.com/seerge/g-helper/files/12475179/GHelper.zip)

**Comment by squee666** [comment]
> @squee666 ok, then brightness 0 is not the right value. Lets try 1 instead of 0
> 
> Check this build [GHelper.zip](https://github.com/seerge/g-helper/files/12475179/GHelper.zip)

nope that puts the computer to sleep

all you need to is take the build below

> @squee666 i don't see video, but check this build
> 
> [GHelper.zip](https://github.com/seerge/g-helper/files/12475045/GHelper.zip)
> 
> `106` logic is
> 
> * Screenpad OFF, Brightness OFF
> * Screenpad ON, Brightness 0%
> * Screenpad ON, Brightness 100%
> * Repeate
> 
> You can check it in a log itself (screenpad on off and brightness are 2 separate commands)
> 
> Screenpad Up / Down buttons logic is - OFF, 0%, 10% ... 90%, 100%


and heres what you need to do swap the logic order from

- Screenpad OFF, Brightness OFF
- Screenpad ON, Brightness 0%
- Screenpad ON, Brightness 100%
- Repeate

change this to 

- Screenpad ON, Brightness 100%
- Screenpad ON, Brightness 0%
- Screenpad OFF, Brightness OFF
- Repeat


this seems to be the requirement for it to function correctly
if you turn off screenpad and turn it back on it will instantly set itself to 100% brightness
then toggling the brightness up drops it down to 20% / 30% etc

**Comment by seerge** [comment]
@squee666 ops, sleep was part of the way i test hotkey (i have G14 so use sleep key to imitate it ...)

Check this build
[GHelper.zip](https://github.com/seerge/g-helper/files/12475227/GHelper.zip)

**Comment by squee666** [comment]
> @squee666 ops, sleep was part of the way i test hotkey (i have G14 so use sleep key to imitate it ...)
> 
> Check this build [GHelper.zip](https://github.com/seerge/g-helper/files/12475227/GHelper.zip)

nope 1% brightness turns the screenpad on but it will default to 100% brightness until i toggle brightness then it well act as if it was set to 1% because the toggle kicks in

seems like Literally needs to be

- Screenpad ON, Brightness 100%
- Screenpad ON, Brightness 0%
- Screenpad OFF, Brightness OFF
- Repeat

or it doesnt function correctly like stock asus

**Comment by seerge** [comment]
@squee666 i don't think that order matters much :)

But i have changed it 
[GHelper.zip](https://github.com/seerge/g-helper/files/12475281/GHelper.zip)

**Comment by squee666** [comment]
> @squee666 i don't think that order matters much :)
> 
> But i have changed it [GHelper.zip](https://github.com/seerge/g-helper/files/12475281/GHelper.zip)
> 
> Does changing brightness with UP / Down commands allow you to set screen to minimum brightness w/o turning it off ?

okay this functions as intended but toggling brightness to 0 turns off display
i dont know if its because you have brightness set to 1 but touch input still functions when that should also disable

**Comment by seerge** [comment]
@squee666 can you check this ?
[GHelper.zip](https://github.com/seerge/g-helper/files/12475364/GHelper.zip)

**Comment by squee666** [comment]
that functions exactly the same as above

**Comment by seerge** [comment]
@squee666 i have set min brightness to 10 in this build instead (instead of 1)

Does it keep display visible in that case ?

[GHelper.zip](https://github.com/seerge/g-helper/files/12475405/GHelper.zip)

**Comment by squee666** [comment]
with 10% brightness set the display is still off
also now touch input is disabled like it should be

so this build functions 100% like asus's software but just ports screenpad at 10% instead of "Screen Pad Hidden"

**Comment by seerge** [comment]
@squee666 you are confusing me completely

If this build still doesn't work as you want 
[GHelper.zip](https://github.com/seerge/g-helper/files/12475605/GHelper.zip)

You will have to pick up parameters manually and tell me them

To Enable / Disable screenpad run in powershell as admin (where 1 / 0 is the status)

```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00050031, 1)
```

To set brightness , where  255 - is brightness value in a range from 0 (min) to 255 (max)

```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00050031, 255)
```

And tell me params that you want for each "mode" :)

**Comment by seerge** [comment]
@squee666 great, closing as completed then
